test: unreliable "job exit event on jobstart(…,{term=true})"

justinmk opened the pull request:
Problem:
Test often fails in cirrus (bsd) ci:

    FAILED   test/functional/terminal/channel_spec.lua @ 149: no crash when TermOpen autocommand processes job exit event when using jobstart(…,{term=true})
    test/functional/terminal/channel_spec.lua:158: Row 1 did not match.
    Expected:
      |*^ready $ foobar                                              |
      |*                                                            |
      |*[Process exited 0]                                          |
      |                                                            |
    Actual:
      |*^ready $                                                     |
      |*[Process exited 0]                                          |
      |*                                                            |
      |                                                            |

Solution:

Use one printf call instead of multiple.

cc @zeertzjq

justinmk committed:
test: unreliable "job exit event on jobstart(…,{term=true})"

Problem:
Test often fails in cirrus (bsd) ci:

    FAILED   test/functional/terminal/channel_spec.lua @ 149: no crash when TermOpen autocommand processes job exit event when using jobstart(…,{term=true})
    test/functional/terminal/channel_spec.lua:158: Row 1 did not match.
    Expected:
      |*^ready $ foobar                                              |
      |*                                                            |
      |*[Process exited 0]                                          |
      |                                                            |
    Actual:
      |*^ready $                                                     |
      |*[Process exited 0]                                          |
      |*                                                            |
      |                                                            |

Solution:
Use one printf call instead of multiple.

zeertzjq reviewed (changes requested):
I don't think this change is correct. The blank line between the `ready $` and `[Process exited 0]` lines will still fail to match.

justinmk committed:
test: unreliable "channel opened, ..., deleted by pressing a key"

Problem:
Test sometimes fails on macos:

    test/functional/terminal/channel_spec.lua:96: Expected objects to be the same.
    Passed in:
    (number) 3
    Expected:
    (number) 2

Solution:
Retry a few times.

justinmk commented on test/functional/terminal/channel_spec.lua:
```diff
@@ -92,7 +92,9 @@ describe('terminal channel is closed and later released if', function()
-    eq(chans - 1, eval('len(nvim_list_chans())'))
+    t.retry(20, nil, function()
+      eq(chans - 1, eval('len(nvim_list_chans())'))
+    end)
```
Seems to improve succeess rate a lot when running locally (macos).

zeertzjq replied:
Ah OK
